src/frontends/tensorflow/src/tf_utils.cpp

```diff
@@ -190,3 +208,146 @@ ov::Any ov::frontend::tensorflow::unpack_tensor_proto(const ::tensorflow::Tensor
     }
     return res;
 }
+
+bool ov::frontend::tensorflow::cf_marker_exists(const shared_ptr<const Node>& node) {
+    const auto& rt_info = node->get_rt_info();
+    if (rt_info.count(cf_marker_tag) > 0) {
+        return true;
+    }
+    return false;
+}
+
+ov::frontend::tensorflow::CfMarkerType ov::frontend::tensorflow::get_cf_marker(const shared_ptr<const Node>& node) {
+    auto rt_info = node->get_rt_info();
+    FRONT_END_GENERAL_CHECK(rt_info.count(cf_marker_tag) > 0,
+                            "[TensorFlow Frontend] internal error: node does not contain conditional flow marker");
+    auto& ov_any_cf_marker = rt_info[cf_marker_tag];
+    FRONT_END_GENERAL_CHECK(ov_any_cf_marker.is<CfMarkerType>(),
+                            "[TensorFlow Frontend] internal error: incorrect type of conditional flow marker");
+    return ov_any_cf_marker.as<CfMarkerType>();
+}
+
+void ov::frontend::tensorflow::set_cf_marker(const ov::frontend::tensorflow::CfMarkerType& cf_marker,
+                                             const std::shared_ptr<ov::Node>& node) {
+    node->get_rt_info()[cf_marker_tag] = cf_marker;
+}
+
+uint32_t ov::frontend::tensorflow::generate_cf_marker() {
```

Comment:
I may suggest to add to a docs information about how it works. Because it looks like as an expected (but not obvious) shared value between sessions. I mean, if you open n+1 model - you shouldn't expect markers restarts from 0.

Reply:
Added.